src/hal/src/pso/descriptor.rs

```diff
@@ -33,38 +33,67 @@ pub type DescriptorBinding = u32;
 ///
 pub type DescriptorArrayIndex = usize;
 
-/// DOC TODO: Grasping and remembering the differences between these
-///       types is a tough task. We might be able to come up with better names?
-///       Or even use tuples to describe functionality instead of coming up with fancy names.
-#[repr(C)]
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+/// Specific type of a buffer.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum BufferDescriptorType {
+    /// Storage buffers allow load, store, and atomic operations.
+    Storage {
+        /// If true, store operations are not permitted on this buffer.
+        read_only: bool,
+    },
+    /// Uniform buffers allow only load operations.
```

Comment:
we should rephrase this, as I don't think users are even expected to call `load` on a uniform buffer (that's what storage buffers are for!).
Maybe "Uniform buffers allow providing constant data to a shader" ?